tables/table.py

```diff
@@ -128,8 +137,13 @@ def restorecache(self):
     self._dirtycache = False
 
 
-def _table__where_indexed(self, compiled, condition, condvars,
-                          start, stop, step):
+def _table__where_indexed(self: "Table",
+                          compiled: "CompiledCondition",
+                          condition: str,
+                          condvars: dict[str, Union["Column", np.ndarray]],
+                          start: int,
+                          stop: int,
+                          step: int) -> Union[Iterator, np.ndarray]:
```

Comment:
See the comment on `__iter__`

Reply:
As recommended changed to `Iterator[tableextension.Row]`.